src/compiler/crystal/codegen/cache_dir.cr

```diff
@@ -119,7 +119,7 @@ module Crystal
         .sort_by! { |dir| File.info?(dir).try(&.modification_time) || Time.unix(0) }
         .reverse!
         .skip(10)
-        .each { |name| `rm -rf "#{name}"` rescue nil }
+        .each { |name| `rm -rf -- #{Process.quote(name)}` rescue nil }
```

Comment:
Why is the `--` needed/introduced here?

Reply:
Oh. Well it's not directly related to the overall theme of quoting, but does fix a security-related issue. If there was a file named just `"-i"`, `rm` would see it as a flag, and the quoting wouldn't help either. I don't think it can actually happen here because cache names are controlled by Crystal, but still good practice to always add the double dash to indicate the end of flags.

Reply:
Why not just do `Dir.delete` instead?

Reply:
Unrelated change

Reply:
I know, so is it adding the `--` then, but I'm not saying you have to do it in this PR. Just asking if there is any reason this was made this way. You were doing the analysis of parts of the code that require `Process.quote` and it could be fixed in a different way instead.

Reply:
I could've split it into a separate commit but they're ignored under the current merging process.
I could've split it into a separate PR, but the process overhead (mainly, but not exclusively, due to GitHub) is too much.
So I just ignore it.

Reply:
So but we could/should have a PR that changes this to `Dir.delete`?

Reply:
Yes, it'd be good. But I am not interested in sending that out. Maybe the whole codebase could be swept for such things.